mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java

```diff
@@ -122,6 +123,83 @@ public StatisticResult addAssists(Transaction transaction, int amount) {
         return new StatisticResult(newValue);
     }
 
+    /**
+     * Adds experience to the user account. Infallible.
+     * @param transaction represents the transaction
+     * @param amount represents the amount of xp to add
+     * @return a result with the new value of xp
+     */
+    public StatisticResult addExperience(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        int existingValue = statisticsRecord.getExperience();
+        int newValue = existingValue + amount;
+
+        statisticsRecord.setExperience(newValue);
+        statisticsRecord.store(KITPVP_STATISTICS.EXPERIENCE);
+
+        this.updateExperience(newValue);
+
+        return new StatisticResult(newValue);
+    }
+
+    /**
+     * Adds current killstreaks to the user account. Infallible.
+     *
+     * This will increment both current and highest killstreaks.
+     *
+     * @param transaction represents the current killstreak
+     * @param amount represents the amount of killstreak to add
+     * @return a result with the new value of killstreak
+     */
+    public StatisticResult addKillstreaks(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        int existingValue = statisticsRecord.getCurrentKillstreak();
+        int newValue = existingValue + amount;
+
+        int existingTwo = statisticsRecord.getHighestKillstreak();
+
+        if (newValue > existingTwo) {
+            int newTwo = existingTwo + amount;
```

Comment:
If amount is ever greater than 1, this is a bug. You need to update the highest killstreak with the new highest killstreak, not the value of the old highest killstreak incremented. Having more meaningful variable names might help with that too

Reply:
what? This code here basically says this: If the old killstreak + 1 is bigger than the existing HIGHEST killstreak, also increment the highest killstreak.

Reply:
If amount is equal to 1, then your code will effectively work fine. If amount is ever greater than 1, the following situation can occur with amount *a*:

* The existing killstreak is k
* The existing highest killstreak is k + 1
* The method is called with amount a
* The new killstreak is k+a
* The new highest killstreak should be k+a, but your code makes it k+1+a
Instead of k + 1, you can use any k + b where b < a.

For example, the player's existing killstreak is 5. Their existing highest killstreak is 7. The method is called with amount 3. The new killstreak is 8. The new highest killstreak should be 8, but your code makes it 10.